src/Libraries/DSIronPython/IronPythonEvaluator.cs

```diff
@@ -272,13 +272,25 @@ public static DataMarshaler OutputMarshaler
         [Obsolete("Do not use! This will be subject to changes in a future version of Dynamo")]
         public static event EvaluationEventHandler EvaluationBegin;
 
+        /// <summary>
+        ///     Emitted immediately before execution begins
+        /// </summary>
+        [SupressImportIntoVM]
+        public static event Action<string, IList, Action<string, object>> EvaluationStarted;
+
         /// <summary>
         ///     Emitted immediately after execution ends or fails
         /// </summary>
         [SupressImportIntoVM]
         [Obsolete("Do not use! This will be subject to changes in a future version of Dynamo")]
```

Comment:
should we change these messages, to note these events are not even raised any longer?